projects/ccf-shared/src/lib/services/globals/tracking.state.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { DataAction, StateRepository } from '@ngxs-labs/data/decorators';
+import { NgxsImmutableDataRepository } from '@ngxs-labs/data/repositories';
+import { State } from '@ngxs/store';
+
+
+export interface PageStateModel {
+  allowTelemetry?: boolean;
+}
+
+export const LOCAL_STORAGE_ALLOW_TELEMETRY_KEY = 'ALLOW_TELEMETRY';
+export const INITIAL_TELEMETRY_SETTING  = localStorage.getItem(LOCAL_STORAGE_ALLOW_TELEMETRY_KEY) === null ? undefined
+  : localStorage.getItem(LOCAL_STORAGE_ALLOW_TELEMETRY_KEY)?.toLowerCase() === 'true';
+
+@StateRepository()
+@State<PageStateModel>({
+  name: 'tracking',
+  defaults: {
+    allowTelemetry: INITIAL_TELEMETRY_SETTING
+  }
+})
+@Injectable()
+export class TrackingState extends NgxsImmutableDataRepository<PageStateModel> {
+
+  @DataAction()
+  setAllowTelemetry(allowTelemetry: boolean): void {
+    localStorage.setItem(LOCAL_STORAGE_ALLOW_TELEMETRY_KEY, allowTelemetry.toString());
+    this.ctx.patchState({ allowTelemetry });
+
+    // This ensures that if telemetry is disabled that it _WONT_ send anything to Google Analytics
```

Comment:
Add logic to not reload if you don't have to. Like when:
1. previous setting is undefined and new setting is true
2. previous setting is the same as new setting